parameter/DomainConfiguration.cpp

```diff
@@ -121,72 +118,74 @@ void CDomainConfiguration::composeSettings(CXmlElement& xmlConfigurationSettings
+        return false;
+    }
 
-            return false;
-        }
-    } else {
+    // Element content
+    CXmlElement xmlConfigurableElementSettingsElementContent;
+    // Check name and kind
+    if (!xmlConfigurableElementSettingsElement.getChildElement(
+                destination->getKind(),
+                destination->getName(),
+                xmlConfigurableElementSettingsElementContent)) {
 
-        // Create child XML element
-        xmlConfigurableElementSettingsElement.createChild(xmlConfigurableElementSettingsElementContent, pConfigurableElement->getKind());
+        context.setError(
+                "Couldn't find settings for " +
+                destination->getKind() + " " + destination->getName() +
+                " for Configuration " + getPath());
 
-        // Set Name
-        xmlConfigurableElementSettingsElementContent.setNameAttribute(pConfigurableElement->getName());
+        return false;
     }
 
-    // Change context type to parameter settings access
-    string strError;
-
     // Create configuration access context
-    CConfigurationAccessContext configurationAccessContext(strError, bSerializeOut);
+    CConfigurationAccessContext configurationAccessContext(context, false);
 
-    // Provide current value space
-    configurationAccessContext.setValueSpaceRaw(xmlDomainExportContext.valueSpaceIsRaw());
+    // Have domain configuration parse settings for configurable element
+    return areaConfiguration->serializeXmlSettings(
+            xmlConfigurableElementSettingsElementContent,
+            configurationAccessContext);
+}
 
-    // Provide current output raw format
-    configurationAccessContext.setOutputRawFormat(xmlDomainExportContext.outputRawFormatIsHex());
+bool CDomainConfiguration::exportOneConfigurableElementSettings(
+        CAreaConfiguration *areaConfiguration,
+        CXmlElement& xmlConfigurableElementSettingsElement,
+        CXmlDomainExportContext& context) const
+{
+    const CConfigurableElement* source = areaConfiguration->getConfigurableElement();
 
-    // Have domain configuration parse settings for configurable element
-    if (!areaConfiguration->serializeXmlSettings(xmlConfigurableElementSettingsElementContent, configurationAccessContext)) {
+    // Create child XML element
+    CXmlElement xmlConfigurableElementSettingsElementContent;
+    xmlConfigurableElementSettingsElement.createChild(xmlConfigurableElementSettingsElementContent, source->getKind());
+    xmlConfigurableElementSettingsElementContent.setNameAttribute(source->getName());
 
-        // Forward error
-        xmlDomainExportContext.setError(strError);
+    // Create configuration access context
+    CConfigurationAccessContext configurationAccessContext(context, true);
+    configurationAccessContext.setValueSpaceRaw(context.valueSpaceIsRaw());
+    configurationAccessContext.setOutputRawFormat(context.outputRawFormatIsHex());
```

Comment:
Should this rawness not be part of a common base class between accessContext and export context... Anyway this is fixed by Francois's rework to make configurationAccessContext part of Serializing context. Will your rework invalidate Francois's ?